### PR TITLE
[ASCII-801] Upgrade go version to 1.20.10

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?
Upgrade go version from 1.20.8 to 1.20.10 
* [release notes](https://go.dev/doc/devel/release#go1.20.minor)

### Describe how to test/QA your changes
Sanity check should be covered by the CI and QA deploy process.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brett0000FF — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
Looks good to me!

File: tasks/update_go.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def update_go(
     """
     import semver
 
-    if not semver.Version.is_valid(version):
+    if not semver.VersionInfo.isvalid(version):
```
Comment (chouquette) [inline tasks/update_go.py:34]: Out of curiosity, is this (and the similar one below) change related to the go update itself?

**Review by chouquette — APPROVED** [review]
LGTM!

Just in case, be sure to remove the `test_only` part before merging :)

**Review by pducolin — APPROVED** [review]

File: tasks/update_go.py (hunk 1 above)
Comment (pgimalac) [reply tasks/update_go.py:40]: I created the task when I did the last Go update, but it seems like I was using the wrong version of `semver` locally...

**Pull request merged**